src/System.Net.Http/tests/UnitTests/HPack/HPackIntegerTest.cs

```diff
@@ -73,5 +73,26 @@ public static IEnumerable<object[]> IntegerCodecRoundTripSamples()
                 }
             }
         }
+
+        [Theory]
+        [InlineData(new byte[] { 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x08 })] // 1 bit too large
+        [InlineData(new byte[] { 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x80 })] // A continuation byte (0x80) where the byte after it would be too large.
+        [InlineData(new byte[] { 0xFF, 0xFF, 0x00 })] // Encoded with 1 byte too many.
+        public void HPack_Integer_TooLarge(byte[] encoded)
+        {
+            Assert.Throws<HPackDecodingException>(() =>
+            {
+                var dec = new IntegerDecoder();
+
+                if (!dec.StartDecode(encoded[0], 7))
```

Comment:
I think this is using StartDecode incorrectly.

From what I can tell, StartDecode assumes that the prefix is always removed before passing in the value (`b`). That seems to be how it's used in practice. We don't actually assert this in StartDecode, but it seems like we should.

Here, you are passing in 0xFF as the encoded value and 7 as the prefix length, which seems like invalid usage. 

Am I missing something here?